tests/unit/acquisition/test_optimizer.py

```diff
@@ -116,6 +155,90 @@ def test_generate_continuous_optimizer_raises_with_invalid_init_params() -> None
+) -> None:
+    search_space = Box([-1, -1], [1, 2])
+    optimizer = generate_continuous_optimizer(
+        num_optimization_runs=num_optimization_runs, num_recovery_runs=num_recovery_runs
+    )
+    with pytest.raises(FailedOptimizationError) as e:
+        optimizer(search_space, _delta_function(10))
+    assert (
+        str(e.value)
+        == f"""
+                    Acquisition function optimization failed,
+                    even after {num_recovery_runs + num_optimization_runs} restarts.
+                    """
+    )
+
+
+@pytest.mark.parametrize("num_optimization_runs", [1, 10])
+@pytest.mark.parametrize("num_initial_samples", [1000, 5000])
+def test_optimize_continuous_uses_correctly_uses_init_params(
+    num_optimization_runs: int, num_initial_samples: int
+) -> None:
+
+    querying_initial_sample = True
+
+    def _target_fn(x: TensorType) -> TensorType:
+        nonlocal querying_initial_sample
+
+        if querying_initial_sample:  # check size of initial sample
+            assert tf.shape(x)[0] == num_initial_samples
+        else:  # check evaluations are in parallel with correct batch size
+            assert tf.shape(x)[0] == num_optimization_runs
+
+        querying_initial_sample = False
+        return _quadratic_sum([0.5, 0.5])(x)
+
+    optimizer = generate_continuous_optimizer(num_initial_samples, num_optimization_runs)
+    optimizer(Box([-1], [1]), _target_fn)
+
+
+@pytest.mark.parametrize("failed_first_optimziation", [True, False])
+@pytest.mark.parametrize("num_recovery_runs", [0, 2, 10])
+def test_optimize_continuous_recovery_runs(
+    failed_first_optimziation: bool, num_recovery_runs: int
+) -> None:
+
+    currently_failing = failed_first_optimziation
+    num_batch_evals = 0
+
+    def _target_fn(x: TensorType) -> TensorType:
+        nonlocal currently_failing
+        nonlocal num_batch_evals
+
+        if (
+            tf.shape(x)[0] > 1
+        ):  # count when batch eval (i.e. random init or when doing recovery runs)
+            num_batch_evals += 1
+
+        if (
+            num_batch_evals > 1
+        ):  # after random init, the next batch eval will be start of recovery run
+            assert (
+                tf.shape(x)[0] == num_recovery_runs
+            )  # check that we do correct number of recovery runs
+            currently_failing = False
+
+        if currently_failing:  # return function that is impossible to optimize
+            return _delta_function(10)
+        else:
+            return _quadratic_sum([0.5, 0.5])  # return function that is easy to optimize
+
+        optimizer = generate_continuous_optimizer(
+            num_optimization_runs=1, num_recovery_runs=num_recovery_runs
+        )
+        if failed_first_optimziation and (num_recovery_runs == 0):
+            with pytest.raises(FailedOptimizationError):
+                optimizer(Box([-1], [1]), _target_fn)
+        else:
+            optimizer(Box([-1], [1]), _target_fn)
```

Comment:
so is there a difference between num_recovery_runs being 2 or 10, we don't check that correct number was really executed?

Reply:
Higher up in that test. We assert that we use the correct number of recovery runs

Reply:
ah indeed, missed that